apiserver/facades/client/application/application_get.go

```diff
@@ -332,10 +341,28 @@ func (c *domainCharm) URL() string {
 	if name == "" {
 		panic(fmt.Sprintf("charm name is empty %+v", c.charm))
 	}
+
+	var arch string
+	switch c.locator.Architecture {
+	case architecture.AMD64:
+		arch = "amd64"
+	case architecture.ARM64:
+		arch = "arm64"
+	case architecture.PPC64EL:
+		arch = "ppc64el"
+	case architecture.S390X:
+		arch = "s390x"
+	case architecture.RISV64:
+		arch = "risv64"
+	default:
+		// If there is no architecture set, we should ignore it.
+	}
```

Comment:
This fixes https://warthogs.atlassian.net/browse/JUJU-7270 - the reason was the charm uploaded, but we couldn't locate the charm when adding the application. Something is incorrect in the client, as the client shouldn't be attempting to come up with a charm URL that wasn't sent from the API server.

Reply:
I think we need to have a method somewhere to reconstruct the charm URL. We can do this in a future PR.

I don't want the Architecture type to have a String() method, we should encode it based on the type.